ClassCastException for Unbinder of subclass

The problem seems to be that the `ScrollableDialog$$ViewBinder` doesn't override `createUnbinder()`. `ScrollableDialog` is a subclass of `SimpleDialog` which is a `FrameLayout`. I'm using 8.0.1 and binding without inheritance works without any issues. 
The stacktrace:
```
Caused by: java.lang.ClassCastException: com.gabrielittner.timetable.app.view.dialog.SimpleDialog$$ViewBinder$InnerUnbinder cannot be cast to com.gabrielittner.timetable.app.view.dialog.ScrollableDialog$$ViewBinder$InnerUnbinder
                at com.gabrielittner.timetable.app.view.dialog.ScrollableDialog$$ViewBinder.bind(ScrollableDialog$$ViewBinder.java:13)
                at com.gabrielittner.timetable.app.view.dialog.ScrollableDialog$$ViewBinder.bind(ScrollableDialog$$ViewBinder.java:10)
                at butterknife.ButterKnife.bind(ButterKnife.java:187)
                at butterknife.ButterKnife.bind(ButterKnife.java:133) 
                at com.gabrielittner.timetable.app.view.dialog.SimpleDialog.onFinishInflate(SimpleDialog.java:81) 
```

The generated ViewBinders:
```
public final class ScrollableDialog$$ViewBinder<T extends ScrollableDialog> extends SimpleDialog$$ViewBinder<T> {
  @Override
  public Unbinder bind(final Finder finder, final T target, Object source) {
    InnerUnbinder unbinder = (InnerUnbinder) super.bind(finder, target, source);
    View view;
    view = finder.findRequiredView(source, 2131755353, "field 'scrollView'");
    target.scrollView = finder.castView(view, 2131755353, "field 'scrollView'");
    view = finder.findRequiredView(source, 2131755349, "field 'dividerTop'");
    target.dividerTop = view;
    view = finder.findRequiredView(source, 2131755351, "field 'dividerBottom'");
    target.dividerBottom = view;
    return unbinder;
  }

  protected static final class InnerUnbinder<T extends ScrollableDialog> extends SimpleDialog$$ViewBinder.InnerUnbinder<T> {
    protected InnerUnbinder(T target) {
      super(target);
    }
  }
}
```
```
public class SimpleDialog$$ViewBinder<T extends SimpleDialog> implements ViewBinder<T> {
  @Override
  public Unbinder bind(final Finder finder, final T target, Object source) {
    InnerUnbinder unbinder = createUnbinder(target);
    View view;
    view = finder.findRequiredView(source, 2131755347, "field 'container'");
    target.container = finder.castView(view, 2131755347, "field 'container'");
    view = finder.findRequiredView(source, 2131755348, "field 'title'");
    target.title = finder.castView(view, 2131755348, "field 'title'");
    view = finder.findRequiredView(source, 2131755352, "field 'spaceBottom'");
    target.spaceBottom = view;
    view = finder.findRequiredView(source, 16908313, "field 'positiveButton'");
    target.positiveButton = finder.castView(view, 16908313, "field 'positiveButton'");
    view = finder.findRequiredView(source, 16908314, "field 'negativeButton'");
    target.negativeButton = finder.castView(view, 16908314, "field 'negativeButton'");
    view = finder.findRequiredView(source, 16908315, "field 'neutralButton'");
    target.neutralButton = finder.castView(view, 16908315, "field 'neutralButton'");
    return unbinder;
  }

  protected InnerUnbinder<T> createUnbinder(T target) {
    return new InnerUnbinder(target);
  }

  protected static class InnerUnbinder<T extends SimpleDialog> implements Unbinder {
    private T target;

    protected InnerUnbinder(T target) {
      this.target = target;
    }

    @Override
    public final void unbind() {
      if (target == null) throw new IllegalStateException("Bindings already cleared.");
      unbind(target);
      target = null;
    }

    protected void unbind(T target) {
      target.container = null;
      target.title = null;
      target.spaceBottom = null;
      target.positiveButton = null;
      target.negativeButton = null;
      target.neutralButton = null;
    }
  }
}
```

--- a/butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java
+++ b/butterknife-compiler/src/main/java/butterknife/compiler/BindingClass.java
@@ -149,7 +149,7 @@ final class BindingClass {
       // Create unbinding class.
       result.addType(createUnbinderClass());
 
-      if (!isFinal) {
+      if (!isFinal || hasParentBinding()) {
         // Now we need to provide child classes to access and override unbinder implementations.
         createUnbinderCreateUnbinderMethod(result);
       }
@@ -182,7 +182,7 @@ final class BindingClass {
     if (!hasParentBinding() || !parentBinding.hasUnbinder()) {
       result.addMethod(createUnbindInterfaceMethod(result));
     }
-    if (!isFinal) {
+    if (!isFinal || hasParentBinding()) {
       result.addMethod(createUnbindMethod(result, generic));
     }
 

--- a/butterknife-compiler/src/test/java/butterknife/BindViewTest.java
+++ b/butterknife-compiler/src/test/java/butterknife/BindViewTest.java
@@ -1,16 +1,15 @@
 package butterknife;
 
+import butterknife.compiler.ButterKnifeProcessor;
 import com.google.common.base.Joiner;
 import com.google.testing.compile.JavaFileObjects;
-
-import org.junit.Test;
-
+import java.util.Arrays;
 import javax.tools.JavaFileObject;
-
-import butterknife.compiler.ButterKnifeProcessor;
+import org.junit.Test;
 
 import static com.google.common.truth.Truth.assertAbout;
 import static com.google.testing.compile.JavaSourceSubjectFactory.javaSource;
+import static com.google.testing.compile.JavaSourcesSubjectFactory.javaSources;
 
 public class BindViewTest {
   @Test public void bindingView() {
@@ -119,6 +118,104 @@ public class BindViewTest {
         .compilesWithoutError()
         .and()
         .generatesSources(expectedSource);
+  }
+
+  @Test public void bindingViewFinalClassWithBaseClass() {
+    JavaFileObject baseSource = JavaFileObjects.forSourceString("test.Base",
+        Joiner.on('\n').join(
+            "package test;",
+            "import android.app.Activity;",
+            "import android.view.View;",
+            "import butterknife.BindView;",
+            "public class Base extends Activity {",
+            "    @BindView(1) View thing;",
+            "}"
+        ));
+    JavaFileObject testSource = JavaFileObjects.forSourceString("test.Test",
+        Joiner.on('\n').join(
+            "package test;",
+            "import android.view.View;",
+            "import butterknife.BindView;",
+            "public final class Test extends Base {",
+            "    @BindView(1) View thing;",
+            "}"
+        ));
+
+    JavaFileObject expectedBaseSource = JavaFileObjects.forSourceString("test/Base$$ViewBinder", ""
+        + "package test;\n"
+        + "import android.view.View;\n"
+        + "import butterknife.Unbinder;\n"
+        + "import butterknife.internal.Finder;\n"
+        + "import butterknife.internal.ViewBinder;\n"
+        + "import java.lang.IllegalStateException;\n"
+        + "import java.lang.Object;\n"
+        + "import java.lang.Override;\n"
+        + "public class Base$$ViewBinder<T extends Base> implements ViewBinder<T> {\n"
+        + "  @Override\n"
+        + "  public Unbinder bind(final Finder finder, final T target, Object source) {\n"
+        + "    InnerUnbinder unbinder = createUnbinder(target);\n"
+        + "    View view;\n"
+        + "    view = finder.findRequiredView(source, 1, \"field 'thing'\");\n"
+        + "    target.thing = view;\n"
+        + "    return unbinder;\n"
+        + "  }\n"
+        + "  protected InnerUnbinder<T> createUnbinder(T target) {\n"
+        + "    return new InnerUnbinder(target);\n"
+        + "  }\n"
+        + "  protected static class InnerUnbinder<T extends Base> implements Unbinder {\n"
+        + "    private T target;\n"
+        + "    protected InnerUnbinder(T target) {\n"
+        + "      this.target = target;\n"
+        + "    }\n"
+        + "    @Override\n"
+        + "    public final void unbind() {\n"
+        + "      if (target == null) throw new IllegalStateException(\"Bindings already cleared.\");\n"
+        + "      unbind(target)\n"
+        + "      target = null;\n"
+        + "    }\n"
+        + "    protected void unbind(T target) {\n"
+        + "      target.thing = null;\n"
+        + "    }\n"
+        + "  }\n"
+        + "}");
+
+    JavaFileObject expectedTestSource = JavaFileObjects.forSourceString("test/Test$$ViewBinder", ""
+        + "package test;\n"
+        + "import android.view.View;\n"
+        + "import butterknife.Unbinder;\n"
+        + "import butterknife.internal.Finder;\n"
+        + "import java.lang.Object;\n"
+        + "import java.lang.Override;\n"
+        + "public final class Test$$ViewBinder<T extends Test> extends Base$$ViewBinder<T> {\n"
+        + "  @Override\n"
+        + "  public Unbinder bind(final Finder finder, final T target, Object source) {\n"
+        + "    InnerUnbinder unbinder = (InnerUnbinder) super.bind(finder, target, source);\n"
+        + "    View view;\n"
+        + "    view = finder.findRequiredView(source, 1, \"field 'thing'\");\n"
+        + "    target.thing = view;\n"
+        + "    return unbinder;\n"
+        + "  }\n"
+        + "  @Override\n"
+        + "  protected InnerUnbinder<T> createUnbinder(T target) {\n"
+        + "    return new InnerUnbinder(target);\n"
+        + "  }\n"
+        + "  protected static final class InnerUnbinder<T extends Test> extends Base$$ViewBinder.InnerUnbinder<T> {\n"
+        + "    protected InnerUnbinder(T target) {\n"
+        + "      super(target);\n"
+        + "    }\n"
+        + "    @Override\n"
+        + "    protected void unbind(T target) {\n"
+        + "      super.unbind(target);\n"
+        + "      target.thing = null;\n"
+        + "    }\n"
+        + "  }\n"
+        + "}");
+
+    assertAbout(javaSources()).that(Arrays.asList(baseSource, testSource))
+        .processedWith(new ButterKnifeProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expectedBaseSource, expectedTestSource);
   }
 
   @Test public void bindingViewInnerClass() {

